src/common/components/body-class-modifier.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { DocumentManipulator } from 'common/document-manipulator';
+import * as React from 'react';
+
+export type BodyClassModifierProps = {
+    classNames: string[];
+    documentManipulator: DocumentManipulator;
+};
+
+// We user our own BodyClassModifier rather than using Helmet's similar functionality because we
+// need multiple BodyClassModifiers to be able to co-exist without overwriting each other or the
+// body classes office fabric maintains.
+export class BodyClassModifier extends React.Component<BodyClassModifierProps> {
+    public render(): JSX.Element {
+        return null;
+    }
+
+    public componentDidMount(): void {
+        this.updateClassNames([], this.props.classNames);
+    }
+
+    public componentDidUpdate(prevProps: BodyClassModifierProps): void {
+        this.updateClassNames(prevProps.classNames, this.props.classNames);
+    }
+
+    public componentWillUnmount(): void {
+        this.updateClassNames(this.props.classNames, []);
+    }
+
+    private updateClassNames(oldClassNames: string[], newClassNames): void {
+        const { documentManipulator } = this.props;
+
+        const addedClassNames = newClassNames.filter(n => !oldClassNames.includes(n));
+        const removedClassNames = oldClassNames.filter(n => !newClassNames.includes(n));
+
+        // document.body may include classes not managed by this BodyClassModifier; leave them as-is
+        let allBodyClassNames = documentManipulator.bodyClassNames;
+        allBodyClassNames = allBodyClassNames.filter(n => !removedClassNames.includes(n));
+        allBodyClassNames.push(...addedClassNames);
+
+        documentManipulator.bodyClassNames = allBodyClassNames;
```

Comment:
This sintax here has always been a little alien to me. basically because `bodyClassNames` is not a normal prop, it's basically a setter here, and the setter does have logic, so this line is not just an assign statement, something else is happening behind the scene (in this case is the call to `css` before actually assigning the class to the document's body).

Would it be easier to read and understand this is not just an assign statement if we have a proper function (instead of a setter)?

Reply:
I'm glad to hear this comment, I debated exactly this question as I was writing it; will switch to the setter method